packages/gitbook/src/components/DocumentView/Table/ViewGrid.tsx

```diff
@@ -7,159 +8,128 @@ import { TableViewProps } from './Table';
 import styles from './table.module.css';
 import { getColumnAlignment } from './utils';
 
+/* Columns are sized in 3 ways:
+     1. Set to auto-size by default, these columns share the available width
+     2. Explicitly set by the user by dragging column separator (we then turn off auto-size)
+     3. Auto-size is turned off without setting a width, we then default to a fixed width of 100px
+*/
 export function ViewGrid(props: TableViewProps<DocumentTableViewGrid>) {
     const { block, view, records, style } = props;
-    const columnsOverThreshold = view.columns.length >= 7;
 
-    const tableWrapper = columnsOverThreshold
-        ? [
-              // has over X columns
-              'overflow-x-auto',
-              'overflow-y-hidden',
-              'mx-auto',
-              'rounded-md',
-              'border',
-              'border-dark/3',
-              'dark:border-light/2',
-          ]
-        : ['overflow-x-auto', 'overflow-y-hidden', 'mx-auto'];
+    /* Calculate how many columns are auto-sized vs fixed width */
+    const columnWidths = view.columnWidths;
+    const autoSizedColumns = view.columns.filter((column) => !columnWidths?.[column]);
+    const fixedColumns = view.columns.filter((column) => columnWidths?.[column]);
 
-    const tableTR = columnsOverThreshold
-        ? ['[&>*+*]:border-l', '[&>*]:px-4']
-        : ['[&>*+*]:border-l', '[&>*+*]:px-4'];
+    const tableWidth = autoSizedColumns.length > 0 ? 'w-full' : 'w-fit';
 
-    const tableTH = columnsOverThreshold ? ['py-3'] : ['py-1', 'pt-0'];
-
-    // Only show the header when configured and not empty
+    /* Only show the header when configured and not empty */
     const withHeader =
         !view.hideHeader &&
         view.columns.some((columnId) => block.data.definition[columnId].title.trim().length > 0);
 
     return (
         <div
-            className={`${tcls(style, 'relative', 'grid', tableWrapper, styles.progressContainer)}`}
+            className={tcls(
+                style,
+                styles.progressContainer,
+                styles.tableWrapper,
+                'relative',
+                'grid',
+                'w-full',
+                'overflow-x-auto',
+                'overflow-y-hidden',
+                'mx-auto',
+            )}
         >
-            {/* ProgressScroller: */}
-            <div
-                className={`${styles.progressOpacitySharp} ${tcls(
```

Comment:
I'm returning this one, but with the new layout model I need to take a closer look + it is a nice to have and not a crucial element to have.

Reply:
We don't need it and it impact DOM sizes, performances (for ex for docs.nexthink.com/platform/user-guide/nexthink-query-language-nql/nql-data-model); it looks like your PR makes it faster this way.